contracts/test/emergency/StarkNetValidator.test.ts

```diff
@@ -25,13 +25,36 @@ describe('StarkNetValidator', function () {
   let starkNetValidator: Contract
   let mockStarkNetMessagingFactory: ContractFactory
   let mockStarkNetMessaging: Contract
+  let mockGasPriceFeed: MockContract
+  let mockAccessController: MockContract
+  let mockAggregator: MockContract
 
   let l2ContractFactory: StarknetContractFactory
   let l2Contract: StarknetContract
 
   before(async () => {
+    if (network.name !== 'hardhat') {
+      // This is so that the network can know about custom errors.
+      // Running against the provided hardhat node does this automatically.
+
+      const buildInfo = await artifacts.getBuildInfo(
+        'src/chainlink/solidity/emergency/StarkNetValidator.sol:StarkNetValidator',
+      )
+      if (!buildInfo) {
+        throw Error('Cannot find build info')
+      }
+      const { solcVersion, input, output } = buildInfo
+      console.log('Sending compilation result for StarkNetValidator test')
+      await network.provider.request({
+        method: 'hardhat_addCompilationResult',
+        params: [solcVersion, input, output],
+      })
+      console.log('Successfully sent compilation result for StarkNetValidator test')
```

Comment:
Let's extract this to an hardhat specific util method that prepares this for us.